### PR TITLE
Use SecurityString in password securely. 

### PR DESCRIPTION
**Pull request by Sandychuang8**
Only convert the SecurityString to string in fleece encoding and password into NetworkCredential.

### CODE REVIEW

File: src/LiteCore/src/LiteCore.Shared/Interop/Fleece.cs [hunk 1]
```diff
@@ -549,6 +551,9 @@ public static void FLEncode(this object obj, FLEncoder* enc)
                 case Blob b:
                     b.FLEncode(enc);
                     break;
+                case SecureString ss:
+                    new NetworkCredential(string.Empty, ss).Password.FLEncode(enc);
```
Comment (Sandychuang8) [inline src/LiteCore/src/LiteCore.Shared/Interop/Fleece.cs:555]: The NetworkCredential class internally always stores the password as a SecureString:
https://referencesource.microsoft.com/#System/net/System/Net/NetworkCredential.cs,125

File: src/Couchbase.Lite.Shared/API/Sync/BasicAuthenticator.cs [hunk 2]
```diff
@@ -50,15 +50,15 @@ public sealed class BasicAuthenticator : Authenticator
...
+        /// [Obsolete("This property is deprecated, please use SecureString PasswordSecureString instead.")]
         [NotNull]
         public string Password { get; }
 
         /// <summary>
         /// Gets the password that this object holds
         /// </summary>
         [NotNull]
-        internal SecureString PasswordSecureString { get; }
+        public SecureString PasswordSecureString { get; }
```
Comment (borrrden) [inline src/Couchbase.Lite.Shared/API/Sync/BasicAuthenticator.cs:61]: Note that the use of SecureString is [no longer recommended by Microsoft](https://github.com/dotnet/platform-compat/blob/master/docs/DE0001.md).  Also this is a public API change, have you consulted @pasin on this?

File: src/LiteCore/src/LiteCore.Shared/Interop/Fleece.cs (hunk 1 above)
Comment (borrrden) [reply src/LiteCore/src/LiteCore.Shared/Interop/Fleece.cs:555]: This ends up in a string in memory anyway so the question is: "Does it really provide more security than before?"

File: src/Couchbase.Lite.Shared/API/Sync/BasicAuthenticator.cs (hunk 2 above)
Comment (pasin) [reply src/Couchbase.Lite.Shared/API/Sync/BasicAuthenticator.cs:61]: Please use what is recommended by Microsoft.

**Review by borrrden — APPROVED** [review]
Discussed offline on the tradeoffs between this and other approaches

**Pull request merged**